### ISSUE TITLE
Error: Failed to parse block

### ISSUE THREAD

**Issue by Ozhiganov** [zone117x/node-cryptonote-pool#163]
Help to understand the problem, when connecting to the pool, the miner issues an error

Error: Failed to parse block
    at Object.BlockTemplate.nextBlob (/home/USER/fork/lib/pool.js:128:23)
    at Object.Miner.getJob (/home/USER/fork/lib/pool.js:302:41)
    at handleMinerMethod (/home/USER/fork/lib/pool.js:531:28)
    at handleMessage (/home/USER/fork/lib/pool.js:660:13)
    at Socket.socket.on.on.on.pushMessage (/home/USER/fork/lib/pool.js:714:25)
    at Socket.emit (events.js:95:17)
    at Socket.<anonymous> (_stream_readable.js:765:14)
    at Socket.emit (events.js:92:17)
    at emitReadable_ (_stream_readable.js:427:10)
    at emitReadable (_stream_readable.js:423:5)

2017-06-07 07:28:44 [master] Pool fork 1 died, spawning replacement worker...

**Comment by xfirm** [comment]
I'm having the same issue, tried code from other forks and scanned through issues and solutions to no avail, really hope there is an easy fix.

**Comment by MelonSmasher** [comment]
Same issue

**Comment by dinho-afsn** [comment]
same issue

**Comment by d3ancole1995** [comment]
same issue

**Comment by dinho-afsn** [comment]
try here https://github.com/zone117x/node-cryptonote-pool/issues/176

Replace in your package.json
"cryptonote-util": "git://github.com/forknote/node-cryptonote-util.git"

Remove the folder node_modules or the module to be replaced
And run npm update again

**Comment by c0urier** [comment]
I have a similar problem:

Basic info:
$USER@host:~/pool> lsb_release -d
Description:    Ubuntu 16.04.3 LTS

$USER@host:~/pool> nodejs -v
v6.11.2

$USER@host:~/pool> npm -v
3.10.10

$USER@host:~/pool> redis-server -v
Redis server v=4.0.1 sha=00000000:0 malloc=jemalloc-4.0.3 bits=64 build=eaeb3715173af01a

The error message happens when a client tries to connect - Everything is starting up correctly.:
Error: Failed to parse block
    at Error (native)
    at Object.nextBlob (/home/$USER/pool/lib/pool.js:128:23)
    at Object.getJob (/home/$USER/pool/lib/pool.js:301:41)
    at handleMinerMethod (/home/$USER/pool/lib/pool.js:510:28)
    at handleMessage (/home/$USER/pool/lib/pool.js:644:13)
    at Socket.<anonymous> (/home/$USER/pool/lib/pool.js:698:25)
    at emitOne (events.js:96:13)
    at Socket.emit (events.js:188:7)
    at readableAddChunk (_stream_readable.js:176:18)
    at Socket.Readable.push (_stream_readable.js:134:10)

The above comment by dinho-afsn did not have any positive effect. Any advice greatly appreciated.

**Comment by mxjoe** [comment]
I have the same problem and the error comes up only when a client connects.

Error: Failed to parse block
    at Error (native)
    at Object.nextBlob (/home/secumine-adm/pool/lib/pool.js:131:23)
    at Object.getJob (/home/secumine-adm/pool/lib/pool.js:305:41)
    at handleMinerMethod (/home/secumine-adm/pool/lib/pool.js:514:28)
    at handleMessage (/home/secumine-adm/pool/lib/pool.js:648:13)
    at Socket.<anonymous> (/home/secumine-adm/pool/lib/pool.js:702:25)
    at emitOne (events.js:96:13)
    at Socket.emit (events.js:188:7)
    at readableAddChunk (_stream_readable.js:176:18)
    at Socket.Readable.push (_stream_readable.js:134:10)

**Comment by d3ancole1995** [comment]
Use NVM (node version manager) and install more 0.10.48. Then remove node_modules folder. Run npm update and your ready to go.

**Comment by mxjoe** [comment]
Thanks d3ancole1995! My XMR pool under [secumine.net](secumine.net) is working now. There where two things I had to adjust:

1. Revert pack in package.json to [git://github.com/Snipa22/node-cryptonote-util#xmr-Nan-2.0](url)
2. Install nodejs v0.10.48, remove the node_modules folder and run npm update.

**Comment by Jjarrard** [comment]
@mxjoe I wish I could hug you!
nvm install v0.10.48 
(you are now using v0.10.48)
rm -rf node_modules
npm update
node init.js
running smooth as silk

**Comment by sitewizardgit** [comment]
Your welcome,

I spent 2 months trying to figure this out

**Comment by dinho-afsn** [comment]
dead project!

**Comment by Jjarrard** [comment]
@dinho-afsn with a few tweaks it still works

**Comment by dinho-afsn** [comment]
besides not working properly you will never be able to make the automatic payments work, I am a programmer and I know what I am talking about.
I have 2 pools currently online

**Comment by Jjarrard** [comment]
Tell that to my automatic payments

**Comment by dinho-afsn** [comment]
because you do not say that it does not work properly, because if you say that your mining payments are working is a lie, another thing that nobody talks about is that every miner that enters the pool using this script will generate a fake miner, so if you have 10 miners will generate 10 more fakes by paying fees

**Comment by dinho-afsn** [comment]
no lies please

**Comment by dinho-afsn** [comment]
I worked in pools for years

**Comment by Jjarrard** [comment]
You should become a life guard

**Comment by dinho-afsn** [comment]
I already am!

**Comment by Jjarrard** [comment]
https://www.reddit.com/r/Monero/comments/6k078b/nodecryptonotepool_automatic_payout_not_working/ all it took to fix autopayments
TLDR: add --disable-rpc-login --trusted-daemon arguments to the monero-wallet-rpc

**Comment by dinho-afsn** [comment]
The problem with this pool is not rpc or something.
I do not recommend using these scripts, it pays to waste time and mount one using packages node js
https://github.com/lucasjones/node-cryptonote-util

**Comment by dinho-afsn** [comment]
please note the staff you developed do not support it. Do you need to say something else?

**Comment by Jjarrard** [comment]
It's a free, open sourced product, I got it working, and it suits my needs

**Comment by RavenHuginn** [comment]
Error: Failed to parse block
    at Object.BlockTemplate.nextBlob (/opt/pool/lib/pool.js:128:23)
    at Object.Miner.getJob (/opt/pool/lib/pool.js:301:41)
    at handleMinerMethod (/opt/pool/lib/pool.js:510:28)
    at handleMessage (/opt/pool/lib/pool.js:644:13)
    at Socket.socket.on.on.on.pushMessage (/opt/pool/lib/pool.js:698:25)
    at Socket.emit (events.js:95:17)
    at Socket.<anonymous> (_stream_readable.js:765:14)
    at Socket.emit (events.js:92:17)
    at emitReadable_ (_stream_readable.js:427:10)
    at emitReadable (_stream_readable.js:423:5)

2017-11-10 12:37:01 [master] Pool fork 1 died, spawning replacement worker...

 "cryptonote-util": "git://github.com/forknote/node-cryptonote-util.git"

node -v
v0.10.48

Not work, help
